### ISSUE TITLE
[Installation] `pip install .[aws]` does not work when installing from source

### ISSUE THREAD

**Issue by romilbhardwaj** [skypilot-org/skypilot#1213]
If you run `pip install .[aws]` in the skypilot repo, it installs skypilot but `import sky` in python or invoking the sky cli fails with `ModuleNotFoundError: No module named 'googleapiclient'`. Installing with `pip install .[aws,gcp]` works.

See full log below.

### Reproduction
1. Create a new VM on aws with ec2linux deep learning AMI
2. ssh into it and run:
```
git clone https://github.com/skypilot-org/skypilot
pip3 install .[aws]
sky --help
``` 
Logs:
```
[ec2-user@ip-172-31-45-145 ~]$ git clone https://github.com/skypilot-org/skypilot
Cloning into 'skypilot'...
remote: Enumerating objects: 11507, done.
remote: Counting objects: 100% (1492/1492), done.
remote: Compressing objects: 100% (658/658), done.
remote: Total 11507 (delta 1060), reused 1190 (delta 833), pack-reused 10015
Receiving objects: 100% (11507/11507), 6.54 MiB | 13.69 MiB/s, done.
Resolving deltas: 100% (8816/8816), done.
[ec2-user@ip-172-31-45-145 ~]$ cd skypilot/
[ec2-user@ip-172-31-45-145 skypilot]$ pip3 install .[aws]
Defaulting to user installation because normal site-packages is not writeable
Processing /home/ec2-user/skypilot
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
Collecting tabulate
  Downloading tabulate-0.9.0-py3-none-any.whl (35 kB)
Collecting Click
  Downloading click-8.1.3-py3-none-any.whl (96 kB)
     |████████████████████████████████| 96 kB 7.1 MB/s
Collecting oauth2client
  Downloading oauth2client-4.1.3-py2.py3-none-any.whl (98 kB)
     |████████████████████████████████| 98 kB 14.0 MB/s
Collecting cryptography
  Downloading cryptography-38.0.1-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)
     |████████████████████████████████| 4.1 MB 86.3 MB/s
Collecting protobuf<4.0.0
  Downloading protobuf-3.20.3-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.0 MB)
     |████████████████████████████████| 1.0 MB 88.5 MB/s
Collecting networkx
  Downloading networkx-2.6.3-py3-none-any.whl (1.9 MB)
     |████████████████████████████████| 1.9 MB 60.6 MB/s
Collecting boto3
  Downloading boto3-1.24.89-py3-none-any.whl (132 kB)
     |████████████████████████████████| 132 kB 83.0 MB/s
Collecting rich
  Downloading rich-12.6.0-py3-none-any.whl (237 kB)
     |████████████████████████████████| 237 kB 79.7 MB/s
Collecting colorama
  Downloading colorama-0.4.5-py2.py3-none-any.whl (16 kB)
Collecting ray[default]<=1.13.0,>=1.9.0
  Downloading ray-1.13.0-cp37-cp37m-manylinux2014_x86_64.whl (54.5 MB)
     |████████████████████████████████| 54.5 MB 10 kB/s
Collecting psutil
  Downloading psutil-5.9.2-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (281 kB)
     |████████████████████████████████| 281 kB 73.9 MB/s
Collecting pycryptodome==3.12.0
  Downloading pycryptodome-3.12.0-cp35-abi3-manylinux2010_x86_64.whl (2.0 MB)
     |████████████████████████████████| 2.0 MB 61.0 MB/s
Collecting jsonschema
  Downloading jsonschema-4.16.0-py3-none-any.whl (83 kB)
     |████████████████████████████████| 83 kB 3.6 MB/s
Collecting pulp
  Downloading PuLP-2.6.0-py3-none-any.whl (14.2 MB)
     |████████████████████████████████| 14.2 MB 59.4 MB/s
Collecting packaging
  Downloading packaging-21.3-py3-none-any.whl (40 kB)
     |████████████████████████████████| 40 kB 9.9 MB/s
Collecting awscli
  Downloading awscli-1.25.90-py3-none-any.whl (3.9 MB)
     |████████████████████████████████| 3.9 MB 59.7 MB/s
Collecting pandas
  Downloading pandas-1.3.5-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.3 MB)
     |████████████████████████████████| 11.3 MB 61.4 MB/s
Collecting PrettyTable
  Downloading prettytable-3.4.1-py3-none-any.whl (26 kB)
Collecting wheel
  Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)
Collecting pendulum
  Downloading pendulum-2.1.2-cp37-cp37m-manylinux1_x86_64.whl (155 kB)
     |████████████████████████████████| 155 kB 81.3 MB/s
Collecting grpcio<=1.43.0
  Downloading grpcio-1.43.0-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)
     |████████████████████████████████| 4.1 MB 38.8 MB/s
Collecting filelock
  Downloading filelock-3.8.0-py3-none-any.whl (10 kB)
Collecting jinja2
  Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)
     |████████████████████████████████| 133 kB 72.7 MB/s
Collecting importlib-metadata; python_version < "3.8"
  Downloading importlib_metadata-5.0.0-py3-none-any.whl (21 kB)
Collecting pyasn1>=0.1.7
  Downloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
     |████████████████████████████████| 77 kB 542 kB/s
Collecting pyasn1-modules>=0.0.5
  Downloading pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
     |████████████████████████████████| 155 kB 88.9 MB/s
Collecting rsa>=3.1.4
  Downloading rsa-4.9-py3-none-any.whl (34 kB)
Collecting six>=1.6.1
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting httplib2>=0.9.1
  Downloading httplib2-0.20.4-py3-none-any.whl (96 kB)
     |████████████████████████████████| 96 kB 679 kB/s
Collecting cffi>=1.12
  Downloading cffi-1.15.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (427 kB)
     |████████████████████████████████| 427 kB 82.8 MB/s
Collecting jmespath<2.0.0,>=0.7.1
  Downloading jmespath-1.0.1-py3-none-any.whl (20 kB)
Collecting botocore<1.28.0,>=1.27.89
  Downloading botocore-1.27.89-py3-none-any.whl (9.2 MB)
     |████████████████████████████████| 9.2 MB 37.5 MB/s
Collecting s3transfer<0.7.0,>=0.6.0
  Downloading s3transfer-0.6.0-py3-none-any.whl (79 kB)
     |████████████████████████████████| 79 kB 18.7 MB/s
Collecting commonmark<0.10.0,>=0.9.0
  Downloading commonmark-0.9.1-py2.py3-none-any.whl (51 kB)
     |████████████████████████████████| 51 kB 565 kB/s
Collecting pygments<3.0.0,>=2.6.0
  Downloading Pygments-2.13.0-py3-none-any.whl (1.1 MB)
     |████████████████████████████████| 1.1 MB 73.2 MB/s
Collecting typing-extensions<5.0,>=4.0.0; python_version < "3.9"
  Downloading typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting requests
  Downloading requests-2.28.1-py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB 3.2 MB/s
Collecting pyyaml
  Downloading PyYAML-6.0-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (596 kB)
     |████████████████████████████████| 596 kB 67.1 MB/s
Collecting frozenlist
  Downloading frozenlist-1.3.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (148 kB)
     |████████████████████████████████| 148 kB 71.5 MB/s
Collecting numpy>=1.16; python_version < "3.9"
  Downloading numpy-1.21.6-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
     |████████████████████████████████| 15.7 MB 35.7 MB/s
Collecting msgpack<2.0.0,>=1.0.0
  Downloading msgpack-1.0.4-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (299 kB)
     |████████████████████████████████| 299 kB 69.0 MB/s
Collecting aiosignal
  Downloading aiosignal-1.2.0-py3-none-any.whl (8.2 kB)
Collecting attrs
  Downloading attrs-22.1.0-py2.py3-none-any.whl (58 kB)
     |████████████████████████████████| 58 kB 13.0 MB/s
Collecting virtualenv
  Downloading virtualenv-20.16.5-py3-none-any.whl (8.8 MB)
     |████████████████████████████████| 8.8 MB 52.1 MB/s
Collecting py-spy>=0.2.0; extra == "default"
  Downloading py_spy-0.3.14-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (3.0 MB)
     |████████████████████████████████| 3.0 MB 41.6 MB/s
Collecting colorful; extra == "default"
  Downloading colorful-0.5.4-py2.py3-none-any.whl (201 kB)
     |████████████████████████████████| 201 kB 89.2 MB/s
Collecting smart-open; extra == "default"
  Downloading smart_open-6.2.0-py3-none-any.whl (58 kB)
     |████████████████████████████████| 58 kB 10.9 MB/s
Collecting aiohttp-cors; extra == "default"
  Downloading aiohttp_cors-0.7.0-py3-none-any.whl (27 kB)
Collecting opencensus; extra == "default"
  Downloading opencensus-0.11.0-py2.py3-none-any.whl (128 kB)
     |████████████████████████████████| 128 kB 71.0 MB/s
Collecting aiohttp>=3.7; extra == "default"
  Downloading aiohttp-3.8.3-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (948 kB)
     |████████████████████████████████| 948 kB 84.7 MB/s
Collecting gpustat>=1.0.0b1; extra == "default"
  Downloading gpustat-1.0.0.tar.gz (90 kB)
     |████████████████████████████████| 90 kB 19.5 MB/s
Collecting prometheus-client<0.14.0,>=0.7.1; extra == "default"
  Downloading prometheus_client-0.13.1-py3-none-any.whl (57 kB)
     |████████████████████████████████| 57 kB 7.4 MB/s
Collecting importlib-resources>=1.4.0; python_version < "3.9"
  Downloading importlib_resources-5.10.0-py3-none-any.whl (34 kB)
Collecting pkgutil-resolve-name>=1.3.10; python_version < "3.9"
  Downloading pkgutil_resolve_name-1.3.10-py3-none-any.whl (4.7 kB)
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Downloading pyrsistent-0.18.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (117 kB)
     |████████████████████████████████| 117 kB 70.3 MB/s
Collecting pyparsing!=3.0.5,>=2.0.2
  Downloading pyparsing-3.0.9-py3-none-any.whl (98 kB)
     |████████████████████████████████| 98 kB 15.5 MB/s
Requirement already satisfied: docutils<0.17,>=0.10 in /usr/lib/python3.7/site-packages (from awscli->skypilot==1.0.0.dev0) (0.14)
Collecting python-dateutil>=2.7.3
  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
     |████████████████████████████████| 247 kB 71.0 MB/s
Collecting pytz>=2017.3
  Downloading pytz-2022.4-py2.py3-none-any.whl (500 kB)
     |████████████████████████████████| 500 kB 81.8 MB/s
Collecting wcwidth
  Downloading wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Collecting pytzdata>=2020.1
  Downloading pytzdata-2020.1-py2.py3-none-any.whl (489 kB)
     |████████████████████████████████| 489 kB 74.0 MB/s
Collecting MarkupSafe>=2.0
  Downloading MarkupSafe-2.1.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting zipp>=0.5
  Downloading zipp-3.9.0-py3-none-any.whl (5.8 kB)
Collecting pycparser
  Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
     |████████████████████████████████| 118 kB 77.5 MB/s
Collecting urllib3<1.27,>=1.25.4
  Downloading urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
     |████████████████████████████████| 140 kB 71.1 MB/s
Collecting certifi>=2017.4.17
  Downloading certifi-2022.9.24-py3-none-any.whl (161 kB)
     |████████████████████████████████| 161 kB 72.0 MB/s
Collecting charset-normalizer<3,>=2
  Downloading charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     |████████████████████████████████| 61 kB 219 kB/s
Collecting platformdirs<3,>=2.4
  Downloading platformdirs-2.5.2-py3-none-any.whl (14 kB)
Collecting distlib<1,>=0.3.5
  Downloading distlib-0.3.6-py2.py3-none-any.whl (468 kB)
     |████████████████████████████████| 468 kB 74.0 MB/s
Collecting google-api-core<3.0.0,>=1.0.0; python_version >= "3.6"
  Downloading google_api_core-2.10.2-py3-none-any.whl (115 kB)
     |████████████████████████████████| 115 kB 89.6 MB/s
Collecting opencensus-context>=0.1.3
  Downloading opencensus_context-0.1.3-py2.py3-none-any.whl (5.1 kB)
Collecting async-timeout<5.0,>=4.0.0a3
  Downloading async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting asynctest==0.13.0; python_version < "3.8"
  Downloading asynctest-0.13.0-py3-none-any.whl (26 kB)
Collecting yarl<2.0,>=1.0
  Downloading yarl-1.8.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (231 kB)
     |████████████████████████████████| 231 kB 81.3 MB/s
Collecting multidict<7.0,>=4.5
  Downloading multidict-6.0.2-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (94 kB)
     |████████████████████████████████| 94 kB 8.1 MB/s
Collecting nvidia-ml-py<=11.495.46,>=11.450.129
  Downloading nvidia_ml_py-11.495.46-py3-none-any.whl (25 kB)
Collecting blessed>=1.17.1
  Downloading blessed-1.19.1-py2.py3-none-any.whl (58 kB)
     |████████████████████████████████| 58 kB 10.9 MB/s
Collecting google-auth<3.0dev,>=1.25.0
  Downloading google_auth-2.12.0-py2.py3-none-any.whl (169 kB)
     |████████████████████████████████| 169 kB 72.9 MB/s
Collecting googleapis-common-protos<2.0dev,>=1.56.2
  Downloading googleapis_common_protos-1.56.4-py2.py3-none-any.whl (211 kB)
     |████████████████████████████████| 211 kB 83.6 MB/s
Collecting cachetools<6.0,>=2.0.0
  Downloading cachetools-5.2.0-py3-none-any.whl (9.3 kB)
Using legacy 'setup.py install' for gpustat, since package 'wheel' is not installed.
Building wheels for collected packages: skypilot
  Building wheel for skypilot (PEP 517) ... done
  Created wheel for skypilot: filename=skypilot-1.0.0.dev0-py3-none-any.whl size=350926 sha256=c139f87bc2ddc44e09046043113bc91bbca016cd818ee0e9e37e5a7dca0fb862
  Stored in directory: /tmp/pip-ephem-wheel-cache-68bp_8nb/wheels/97/cf/4f/f76de461b63f7865add6d8885cf50596ff5ea95d3fe6f3d65d
Successfully built skypilot
Installing collected packages: tabulate, zipp, typing-extensions, importlib-metadata, Click, pyasn1, pyasn1-modules, rsa, six, pyparsing, httplib2, oauth2client, pycparser, cffi, cryptography, protobuf, networkx, jmespath, python-dateutil, urllib3, botocore, s3transfer, boto3, commonmark, pygments, rich, colorama, certifi, charset-normalizer, idna, requests, pyyaml, frozenlist, importlib-resources, pkgutil-resolve-name, attrs, pyrsistent, jsonschema, numpy, msgpack, aiosignal, platformdirs, distlib, filelock, virtualenv, grpcio, py-spy, colorful, smart-open, async-timeout, asynctest, multidict, yarl, aiohttp, aiohttp-cors, cachetools, google-auth, googleapis-common-protos, google-api-core, opencensus-context, opencensus, nvidia-ml-py, psutil, wcwidth, blessed, gpustat, prometheus-client, ray, pycryptodome, pulp, packaging, awscli, pytz, pandas, PrettyTable, wheel, pytzdata, pendulum, MarkupSafe, jinja2, skypilot

    Running setup.py install for gpustat ... done

ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

ray 1.13.0 requires click<=8.0.4,>=7.0, but you'll have click 8.1.3 which is incompatible.
awscli 1.25.90 requires colorama<0.4.5,>=0.2.5, but you'll have colorama 0.4.5 which is incompatible.
awscli 1.25.90 requires PyYAML<5.5,>=3.10, but you'll have pyyaml 6.0 which is incompatible.
awscli 1.25.90 requires rsa<4.8,>=3.1.2, but you'll have rsa 4.9 which is incompatible.
Successfully installed Click-8.1.3 MarkupSafe-2.1.1 PrettyTable-3.4.1 aiohttp-3.8.3 aiohttp-cors-0.7.0 aiosignal-1.2.0 async-timeout-4.0.2 asynctest-0.13.0 attrs-22.1.0 awscli-1.25.90 blessed-1.19.1 boto3-1.24.89 botocore-1.27.89 cachetools-5.2.0 certifi-2022.9.24 cffi-1.15.1 charset-normalizer-2.1.1 colorama-0.4.5 colorful-0.5.4 commonmark-0.9.1 cryptography-38.0.1 distlib-0.3.6 filelock-3.8.0 frozenlist-1.3.1 google-api-core-2.10.2 google-auth-2.12.0 googleapis-common-protos-1.56.4 gpustat-1.0.0 grpcio-1.43.0 httplib2-0.20.4 idna-3.4 importlib-metadata-5.0.0 importlib-resources-5.10.0 jinja2-3.1.2 jmespath-1.0.1 jsonschema-4.16.0 msgpack-1.0.4 multidict-6.0.2 networkx-2.6.3 numpy-1.21.6 nvidia-ml-py-11.495.46 oauth2client-4.1.3 opencensus-0.11.0 opencensus-context-0.1.3 packaging-21.3 pandas-1.3.5 pendulum-2.1.2 pkgutil-resolve-name-1.3.10 platformdirs-2.5.2 prometheus-client-0.13.1 protobuf-3.20.3 psutil-5.9.2 pulp-2.6.0 py-spy-0.3.14 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycparser-2.21 pycryptodome-3.12.0 pygments-2.13.0 pyparsing-3.0.9 pyrsistent-0.18.1 python-dateutil-2.8.2 pytz-2022.4 pytzdata-2020.1 pyyaml-6.0 ray-1.13.0 requests-2.28.1 rich-12.6.0 rsa-4.9 s3transfer-0.6.0 six-1.16.0 skypilot-1.0.0.dev0 smart-open-6.2.0 tabulate-0.9.0 typing-extensions-4.4.0 urllib3-1.26.12 virtualenv-20.16.5 wcwidth-0.2.5 wheel-0.37.1 yarl-1.8.1 zipp-3.9.0
[ec2-user@ip-172-31-45-145 skypilot]$ python3
Python 3.7.10 (default, Jun  3 2021, 00:02:01)
[GCC 7.3.1 20180712 (Red Hat 7.3.1-13)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sky
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ec2-user/skypilot/sky/__init__.py", line 10, in <module>
    from sky import backends
  File "/home/ec2-user/skypilot/sky/backends/__init__.py", line 3, in <module>
    from sky.backends.cloud_vm_ray_backend import CloudVmRayBackend
  File "/home/ec2-user/skypilot/sky/backends/cloud_vm_ray_backend.py", line 24, in <module>
    from sky import cloud_stores
  File "/home/ec2-user/skypilot/sky/cloud_stores.py", line 14, in <module>
    from sky.data import data_utils
  File "/home/ec2-user/skypilot/sky/data/__init__.py", line 2, in <module>
    from sky.data.storage import Storage, StoreType
  File "/home/ec2-user/skypilot/sky/data/storage.py", line 12, in <module>
    from sky.backends import backend_utils
  File "/home/ec2-user/skypilot/sky/backends/backend_utils.py", line 36, in <module>
    from sky import authentication as auth
  File "/home/ec2-user/skypilot/sky/authentication.py", line 17, in <module>
    import googleapiclient
ModuleNotFoundError: No module named 'googleapiclient'
```

**Comment by concretevitamin** [comment]
This is being addressed by @ewzeng in https://github.com/skypilot-org/skypilot/pull/1209.

**Comment by concretevitamin** [comment]
Merged #1209. This should’ve been fixed now.
